docs/en/week14/week14-1.md

```diff
@@ -0,0 +1,185 @@
+
+<img src="https://i.imgur.com/T7ajVO7.png" style="zoom:100%;"/>
+
+This can be visualized as a matrix by arrranging the feature vectors from TDNN horizontally and the word templates vertically. Each entry in the matrix corresponds to the distance between the feature vector. This can be visualized as a graph problem where the aim is to start from the bottom left-hand corner and reach the top right corner by traversing the path that minimizes the distance.
+
+To train this latent variable model we need to make the energy for the correct answers as small as possible and larger for every correct answer. To do this we use an objective function that takes in templates for wrong words and pushes them away from the current sequence of features and backpropagates the gradients.
+
+## Energy based factor graphs
+The idea behind energy-based factor graphs is to build an energy-based model in which the energy is sum of partial energy terms or when the probability is a product of factors. The benefit of these models is that efficient inference algorithms can be employed.
+
+<img src="https://i.imgur.com/5YE5pog.png" style="zoom:100%;"/>
+
+### Sequence Labeling
+The model takes an input speech signal X and output the labels Y such that the output labels minimize the total energy term.
+<img src="https://i.imgur.com/9k4Frq1.png" style="zoom:100%;"/>
+
+<img src="https://i.imgur.com/1k5oVI3.png" style="zoom:100%;"/>
+
+In this case, the energy is a sum of three terms represented by blue squares which are neural networks that produce feature vectors for the input variables. In the case of speech recognition X can be thought of as a speech signal and the squares implement the grammatical constraints and Y represent the generated output labels.
+<!-- RISHABH --------->
+
+<!--- done by Muge --->
+
+## Efficient Inference for Energy-Based Factor Graphs
+
+A Tutorial on Energy-Based Learning (Yann LeCun, Sumit Chopra, Raia Hadsell, Marc’Aurelio Ranzato, and Fu Jie Huang 2006):
+
+Learning and inference with Energy-Based Models involves a minimization of the energy over the set of answers $\mathcal{Y}$ and latent variables $\mathcal{Z}$. When the cardinality of $\mathcal{Y}\times \mathcal{Z}$ is large, this minimization can become intractable. One approach to the problem is to exploit the structure of the energy function in order to perform the minimization efficiently. One case where the structure can be exploited occurs when the energy can be expressed as a sum of individual functions (called factors) that each depend on different subsets of the variables in Y and Z. These dependencies are best expressed in the form of a factor graph. Factor graphs are a general form of graphical models, or belief networks.
+
+<img src="https://i.imgur.com/I56iwGa.png" style="zoom:100%;"/>
```

Comment:
Images have to be added to the repo. Check out the instructions!
https://atcold.github.io/pytorch-Deep-Learning/en/about/

Reply:
Hi Alf, 
Sorry I didn't get your point. I can see the image on HackMD, what do you mean add images to the repo? Besides, should 14.md be written by 3 editors for different parts? And how to update _config.yml? Thank you!

Reply:
> Images have to be added to the repo. Check out the instructions!
> https://atcold.github.io/pytorch-Deep-Learning/en/about/

Hi Alf,
Sorry I didn't get your point. I can see the image on HackMD, what do you mean add images to the repo? Besides, should 14.md be written by 3 editors for different parts? And how to update _config.yml? Thank you!

Reply:
Images have to be uploaded to `docs/images/week14/14-1`. Check the syntax at the link I've sent you. You can also check out previous PRs.

Reply:
> Images have to be uploaded to `docs/images/week14/14-1`. Check the syntax at the link I've sent you. You can also check out previous PRs.

Hi Alf,

I have uploaded all the images.